index/repo.go

```diff
@@ -0,0 +1,50 @@
+package index
+
+import (
+	"github.com/filecoin-project/dagstore/shard"
+)
+
+// Repo is the central index repository object that manages full indices and
+// manifests.
+type Repo struct {
+	FullIndexRepo
+	ManifestRepo
+}
+
+type Stat struct {
+	Exists bool
```

Comment:
Question: When is Exists false?

Reply:
When the index exists on disk. This may be a little bit of YAGNI FWIW, but useful for testing. Feel free to panic in the implementation when `Stat` is called now. Then we can decide if we do need it or not.